### PR TITLE
Added scala quick start example

### PR DESCRIPTION
**Pull request by pranavanand**
- Added a scala quick start example to simulate the quick start guide.
 - Created sbt project to easily run the quick start guide.

### CODE REVIEW

**Comment by pranavanand** [comment]
@tdas

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      .writeStream
+      .format("console")
+      .start()
+
+    stream2.awaitTermination(10000)
+    stream2.stop()
+
+    // Function to upsert microBatchOutputDF into Delta Lake table using merge
+    println("Streaming upgrades in update mode")
+    def upsertToDelta(microBatchOutputDF: DataFrame, batchId: Long) {
```
Comment (tdas) [inline examples/scala/src/main/scala/example/Streaming.scala]: Update this in the same way @rahulsmahadev updated his example to use "key" and "value"

File: examples/scala/src/main/scala/example/Utilities.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.commons.io.FileUtils
+
+import org.apache.spark.sql.SparkSession
+
+object Utilities {
+  def main(args: Array[String]): Unit = {
+    // Create a Spark Session with SQL enabled
+    val spark = SparkSession
+      .builder()
+      .appName("...")
```
Comment (tdas) [inline examples/scala/src/main/scala/example/Utilities.scala]: dont use "..." ... thats a bad pattern for people to copy paste.

File: examples/scala/src/main/scala/example/Utilities.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      .master("local[4]")
+    // config io.delta.sql.DeltaSparkSessionExtension - to enable custom Delta-specific SQL commands
```
Comment (tdas) [inline examples/scala/src/main/scala/example/Utilities.scala]: indent

File: examples/scala/build.sbt [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
```
Comment (rahulsmahadev) [inline examples/scala/build.sbt]: Do we need the ASF license here ? @tdas

Comment (pranavanand) [reply examples/scala/build.sbt]: Ah just checked the delta build.sbt and it doesnt have it will remove that one

**Comment by databricks-cla-assistant** [comment]
[CLA assistant check](https://cla.dev.databricks.com/delta-io/delta?pullRequest=201) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.databricks.com/delta-io/delta?pullRequest=201) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dev.databricks.com/check/delta-io/delta?pullRequest=201) it.</sub>

**Comment by tdas** [comment]
LGTM.

**Pull request closed without merging**